middleware/validateRoomObjectId.js

```diff
@@ -0,0 +1,12 @@
+const errorMessage = require('../utils/errorMessage');
+
+const validateRoomObjectId = (req, res, next) => {
+  const { _roomId } = req.params;
```

Comment:
Please update the function in `validateObjectId.js` to:
`const _id = req.params._roomId || req.params._id;` 
so we can reuse that one and keep only one function for ID validation.

Reply:
Changing this to generalise subDoc function as discussed.